### PR TITLE
Adds Section About Release Branches and Backports

### PR DESCRIPTION
**Pull request by Weltraumschaf**

### CODE REVIEW

File: docs/backports/backports.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+- you have to figure out which _release branches_ are affected by the bug.
+
+One good practice is to initially fix the bug int the branch where it occurred. Also, a good practice is to first implement a _regression test_ which provokes the bug and fail initially. It is highly recommended to add this tests in a separate commits. Now the process is to go back all maintained older version (checkout their _Release branch_) and verify if the bug is there. Simply by looking at the source code, if it's a simple bug or by chery-picking the regression test and execute it. If the branch is affected then cherry-pick the bugfix commits. Then go through all newer release branches until up to the _main branch_ and do the same. (This is also called backporting, although it does not go backward but forward.)
+
+:::note 
+It's quite debatable if this approach is the best. The author of this site made good experience with this approach. Anyway whatever approach is used at the end, one thing is important: Define one approach and always stick to it. Otherwise, you will lose the overview where to fix the bug and where it is already fixed.
```
Comment (Ilyesbdlala) [inline docs/backports/backports.md:141]: "The author of this site" phrasing seems weird, since it's a mainly a documentation website. I would go with the simple scientific-paper style "we". Or you can simply name yourself at the end of the post as the author.

File: docs/backports/backports.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+Also, it is not always possible to simply cherry-pick a bugfix. This may have different reasons:
+
+- The commit does not only contain the bugfix, but also some refactorings or cleanups: **NEVER DO THIS**! For such "hot fixes" bluntly implement the sole bugfix, nothing else. If you want to do refactoring or cleanups? Fine, but do this in different commits. Unless the person who must do the backporting will hate you 😉
```
Comment (Ilyesbdlala) [inline docs/backports/backports.md:145]: " If you want to do refactoring or cleanups? Fine, but do this in different commits. Unless the person who must do the backporting will hate you 😉 "
 Sorry 😅

File: docs/backports/backports.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+- the bug may not exist in _main head version_ because it was fixed by accident or
+- you did massive refactorings which makes simple cherry-picking into older _release branches_ impossible and
+- you have to figure out which _release branches_ are affected by the bug.
+
+One good practice is to initially fix the bug int the branch where it occurred. Also, a good practice is to first implement a _regression test_ which provokes the bug and fail initially. It is highly recommended to add this tests in a separate commits. Now the process is to go back all maintained older version (checkout their _Release branch_) and verify if the bug is there. Simply by looking at the source code, if it's a simple bug or by chery-picking the regression test and execute it. If the branch is affected then cherry-pick the bugfix commits. Then go through all newer release branches until up to the _main branch_ and do the same. (This is also called backporting, although it does not go backward but forward.)
```
Comment (Ilyesbdlala) [inline docs/backports/backports.md:138]:
```suggestion
One good practice is to initially fix the bug int the branch where it occurred. Also, a good practice is to first implement a _regression test_ which provokes the bug and fail initially. It is highly recommended to add this tests in a separate commits. Now the process is to go back all maintained older version (checkout their _Release branch_) and verify if the bug is there. Simply by looking at the source code, if it's a simple bug or by cherry-picking the regression test and execute it. If the branch is affected then cherry-pick the bugfix commits. Then go through all newer release branches until up to the _main branch_ and do the same. (This is also called backporting, although it does not go backward but forward.)
```

**Review by Ilyesbdlala — CHANGES_REQUESTED** [review]

**Comment by Ilyesbdlala** [comment]
"Rewriting this, because it was not included in the review somehow"

I loved it, it was a nice read :) I was actually wondering why the fix should be included in the release branch first, only to find out it's the next section ^_^
Suggested some typo fixes and small changes. It also seems to be written in the style of a blogpost, some maybe include it in the blog instead ? But i also see the usefulness of having it in the docs. So your call.
Otherwise, for someone who had only passing knowledge of backporting, it was easy to understand.
LGTM.

File: docs/backports/backports.md (hunk 1 above)
Comment (Weltraumschaf) [reply docs/backports/backports.md]: My intention was: We as a team have no experience with that. But doesn't really matter.

File: docs/backports/backports.md (hunk 2 above)
Comment (Weltraumschaf) [reply docs/backports/backports.md:145]: Tried to be funny :-) Is it ok or any suggestion?

**Comment by Weltraumschaf** [comment]
I've tried to write it in a less boring way. But of course the whole text can be improved. I think we should improve any text in the docs from time to time. My experience is: It helps to read it in some weeks/months again, and then refine it.

File: docs/backports/backports.md (hunk 2 above)
Comment (Ilyesbdlala) [reply docs/backports/backports.md:145]: Yeah, it's okay. I also tried to be funny ^^

**Review by Ilyesbdlala — APPROVED** [review]
